backend/json_test.go

```diff
@@ -43,14 +45,90 @@ func TestResponseEncoder(t *testing.T) {
 	require.Equal(t, str, string(b2), "same result from pointer or object")
 
 	// Now the same thing in query data
-	qdr := NewQueryDataResponse()
+	qdr := backend.NewQueryDataResponse()
 	qdr.Responses["A"] = dr
 
 	b, err = json.Marshal(qdr)
 	require.NoError(t, err)
 
 	str = string(b)
-	require.Equal(t, `{"responses":{"A":{"frames":[{"schema":{"name":"simple","fields":[{"name":"time","type":"time","typeInfo":{"frame":"time.Time"}},{"name":"valid","type":"bool","typeInfo":{"frame":"bool"}}]},"data":{"values":[[1577934240000,1577934300000],[true,false]]}},{"schema":{"name":"other","fields":[{"name":"value","type":"number","typeInfo":{"frame":"float64"}}]},"data":{"values":[[1]]}}]}}}`, str)
```

Comment:
Not sure I understand a motivation of this change - the map format should be enough for a frontend to consume responses in a desired order. I see that the code you removed from a client-side has some additional logic and complications - but just can't understand why we can't do sth like this:

```
// client
queries = [A, B, C]
responses = giveMeResponsesFor(queries)
for q in queries:
  response = responses[q]
```

What I am missing here?

Reply:
:thinking: you may be right

The existing code had logic to make sure all results (even ones that do not exist in the queries) get returned.  This is likely more complex than it should be -- this may be trying to solve/support no-longer-used setups

Reply:
Maybe it would be much simpler to refactor client-side code a bit to be more compact and maybe remove some non-actual craft from it - looks like the only thing it does is adding unknown queries to final result too if the amount of refIds in query and response does not match (which I believe does not happen too often and does not deserve a new response format from a backend)? It also looks at some `series` field - looks like we don't have one anymore?

Current response format seems pretty sufficient.